### ISSUE TITLE
Better "bootstrap experience"

### ISSUE THREAD

**Issue by dgageot** [GoogleContainerTools/jib#1251]
This issue sums up some feedback shared by @arun-gupta:

He likes Jib a lot. He also thinks the “bootstrap experience” can be improved a lot.
What can improve to make jib work better the first time someone’s using it.

Issues he shared are:
 + Why do we have to install a credential helper if all we do is push to the docker hub and everything is properly setup in ~/.docker/config.json?
 + Can we print some feedback on progress when an image is being pushed? Especially the first time since it’s obviously going to take longer than subsequent times.
 + Should the image really be pushed by default? Let’s say you are trying Jib without the auth being configured, it’s really annoying that the build is failing.

**Comment by arun-gupta** [comment]
That summarizes the experience very well. Thanks @dgageot 

Wondering if @chanezon has played with this and have ideas on how to simplify the `docker login` experience?

**Comment by chanseokoh** [comment]
Thanks for the feedback.

You don't need to install a cred helper if everything is set up in ~/.docker/config.json. If it is not working, it is a bug.

Regarding the push, we should consider that a lot of people are using Jib primiarly because there is no docker installed locally (or in a CI/CD pipeline).

**Comment by briandealwis** [comment]
You shouldn't need to do anything for Docker Hub.  I tried from a dummy account:

```
$ docker logout
Warning: failed to get default registry endpoint from daemon (Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get http://%2Fvar%2Frun%2Fdocker.sock/v1.38/info: dial unix /var/run/docker.sock: connect: permission denied). Using system default: https://index.docker.io/v1/
Not logged in to https://index.docker.io/v1/

$ ls ~/.docker
ls: /Users/dummy/.docker: No such file or directory

$ cd examples/hellowworld

$ git diff
diff --git a/examples/helloworld/pom.xml b/examples/helloworld/pom.xml
index 22e8e3c..60bd911 100644
--- a/examples/helloworld/pom.xml
+++ b/examples/helloworld/pom.xml
@@ -37,6 +37,7 @@
         <artifactId>jib-maven-plugin</artifactId>
         <version>${jib-maven-plugin.version}</version>
         <configuration>
+         <from><image>openjdk:11-slim</image></from>
           <to>
             <image>gcr.io/REPLACE-WITH-YOUR-GCP-PROJECT/image-built-with-jib</image>
           </to>

$ mvn compile jib:buildTar -Dimage=jdk11based
[INFO] Scanning for projects...
[INFO] 
[INFO] -------------------------< example:helloworld >-------------------------
[INFO] Building helloworld 1
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ helloworld ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ helloworld ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- jib-maven-plugin:0.10.0:buildTar (default-cli) @ helloworld ---
[INFO] 
[INFO] Containerizing application to file at '/private/tmp/jib/examples/helloworld/target/jib-image.tar'...
[INFO] Getting base image openjdk:11-slim...
[INFO] Building dependencies layer...
[INFO] Building resources layer...
[INFO] Building classes layer...
[INFO] The base image requires auth. Trying again for openjdk:11-slim...
[INFO] Retrieving registry credentials for registry.hub.docker.com...
[INFO] Finalizing...
[INFO] 
[INFO] Container entrypoint set to [java, -cp, /app/resources:/app/classes:/app/libs/*, example.HelloWorld]
[INFO] Building image to tar file...
[INFO] 
[INFO] Built image tarball at /private/tmp/jib/examples/helloworld/target/jib-image.tar
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 35.990 s
[INFO] Finished at: 2018-11-15T08:57:21-05:00
[INFO] ------------------------------------------------------------------------
```

The message you see about _auth_ is about authorization: registry.hub.docker.com requires fetching an authorization token even to pull.

**Comment by chanseokoh** [comment]
And the push to Docker hub should also work with ~/.docker/config.json without a cred helper.

**Comment by briandealwis** [comment]
> Should the image really be pushed by default? Let’s say you are trying Jib without the auth being configured, it’s really annoying that the build is failing.

We do offer `jib:dockerBuild` to build to the configured docker daemon and `jib:buildTar` to build to a tar file.

**Comment by briandealwis** [comment]
Provding progress on pulling and pushing is definitely an area for improvement.  The experience of pushing to the same registry would be faster with #169.

**Comment by briandealwis** [comment]
@arun-gupta could you share the contents of your `~/.docker/config.json`?

**Comment by chanezon** [comment]
@arun-gupta sorry I haven't tried it yet.

**Comment by coollog** [comment]
UPDATE: We've released `v1.0.0-rc1` with added progress bar feedback during the build!
